### DISCUSSION TITLE
Rag broken in 0.3.30 - collection errors

### DISCUSSION THREAD

**Discussion by arsaboo** [open-webui/open-webui#5743] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.30
- **Ollama (if applicable):** 0.3.12

- **Operating System:** Ubuntu
- **Browser (if applicable):** Chrome

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

Rag was working until recently. But now, it stopped working:
![image](https://github.com/user-attachments/assets/973b0107-33c7-480c-a178-3198b5780da0)

I keep seeing these two errors in the logs:

```bash
ERROR [open_webui.apps.rag.utils] Error when querying the collection: Collection 42d66e301b53a9859cb98ba3066dd30477bfc5ed6f97e2222bc7ec8e0920846 does not exist.
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/rag/utils.py", line 191, in query_collection
    result = query_doc(
             ^^^^^^^^^^
  File "/app/backend/open_webui/apps/rag/utils.py", line 85, in query_doc
    raise e
  File "/app/backend/open_webui/apps/rag/utils.py", line 73, in query_doc
    result = VECTOR_DB_CLIENT.search(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/rag/vector/dbs/chroma.py", line 52, in search
    collection = self.client.get_collection(name=collection_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/api/client.py", line 142, in get_collection
    model = self._server.get_collection(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/telemetry/opentelemetry/__init__.py", line 146, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/api/fastapi.py", line 226, in get_collection
    resp_json = self._make_request(
                ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/api/fastapi.py", line 87, in _make_request
    BaseHTTPClient._raise_chroma_error(response)
  File "/usr/local/lib/python3.11/site-packages/chromadb/api/base_http_client.py", line 92, in _raise_chroma_error
    raise chroma_error
chromadb.errors.InvalidCollectionException: Collection 42d66e301b53a9859cb98ba3066dd30477bfc5ed6f97e2222bc7ec8e0920846 does not exist.
INFO  [open_webui.apps.ollama.main] url: http://host.docker.internal:11434
Collection 42d66e301b53a9859cb98ba3066dd30477bfc5ed6f97e2222bc7ec8e0920846 does not exist.
```

```bash
ERROR [open_webui.apps.rag.utils] Error when querying the collection: Collection.__init__() got an unexpected keyword argument 'log_position'
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/rag/utils.py", line 191, in query_collection
    result = query_doc(
             ^^^^^^^^^^
  File "/app/backend/open_webui/apps/rag/utils.py", line 85, in query_doc
    raise e
  File "/app/backend/open_webui/apps/rag/utils.py", line 73, in query_doc
    result = VECTOR_DB_CLIENT.search(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/rag/vector/dbs/chroma.py", line 52, in search
    collection = self.client.get_collection(name=collection_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/api/client.py", line 142, in get_collection
    model = self._server.get_collection(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/telemetry/opentelemetry/__init__.py", line 146, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/api/fastapi.py", line 232, in get_collection
    model = CollectionModel.from_json(resp_json)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/chromadb/types.py", line 143, in from_json
    return cls(configuration=configuration, **params_map)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Collection.__init__() got an unexpected keyword argument 'log_position'
INFO  [open_webui.apps.ollama.main] url: http://host.docker.internal:11434
Collection.__init__() got an unexpected keyword argument 'log_position'
```

Happy to provide any additional details.

**Comment by arsaboo** [comment]
RAG works fine if I remove chromaDB

**Reply by Kryszn0** [reply]
Similar problem and identical "fix" (removing chromaDB), only difference in my setup is I also have Tika running

**Comment by mbelur** [comment]
I see the same with milvus db as well.

> ERROR [open_webui.apps.rag.utils] Error when querying the collection: <MilvusException: (code=101, message=failed to search: collection not loaded[collection=452817282104588693])>
Traceback (most recent call last):
  File "/app/backend/open_webui/apps/rag/utils.py", line 191, in query_collection
    result = query_doc(
             ^^^^^^^^^^
  File "/app/backend/open_webui/apps/rag/utils.py", line 85, in query_doc
    raise e
  File "/app/backend/open_webui/apps/rag/utils.py", line 73, in query_doc
    result = VECTOR_DB_CLIENT.search(
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/rag/vector/dbs/milvus.py", line 130, in search
    result = self.client.search(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymilvus/milvus_client/milvus_client.py", line 336, in search
    raise ex from ex
  File "/usr/local/lib/python3.11/site-packages/pymilvus/milvus_client/milvus_client.py", line 322, in search
    res = conn.search(
          ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymilvus/decorators.py", line 148, in handler
    raise e from e
  File "/usr/local/lib/python3.11/site-packages/pymilvus/decorators.py", line 144, in handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymilvus/decorators.py", line 183, in handler
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymilvus/decorators.py", line 123, in handler
    raise e from e
  File "/usr/local/lib/python3.11/site-packages/pymilvus/decorators.py", line 87, in handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymilvus/client/grpc_handler.py", line 800, in search
    return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymilvus/client/grpc_handler.py", line 741, in _execute_search
    raise e from e
  File "/usr/local/lib/python3.11/site-packages/pymilvus/client/grpc_handler.py", line 734, in _execute_search
    check_status(response.status)
  File "/usr/local/lib/python3.11/site-packages/pymilvus/client/utils.py", line 63, in check_status
    raise MilvusException(status.code, status.reason, status.error_code)
pymilvus.exceptions.MilvusException: <MilvusException: (code=101, message=failed to search: collection not loaded[collection=452817282104588693])>

**Comment by arsaboo** [comment]
@mbelur well, that is good. At least the error is reproducible. Hopefully, someone can get to it.

**Comment by bgeneto** [comment]
RAG works fine for me in dev branch, better yet after #5728 :-)

**Comment by arsaboo** [comment]
@bgeneto are you using any of the external DBs?

**Comment by bgeneto** [comment]
> @bgeneto are you using any of the external DBs?

I prefer using Chroma DB, but tested with Chroma, Milvus, Ollama, and the built-in sentence transformer. Best results for me so far was with `jina-embeddings-v3` + `mxbai-rerank-xsmall-v1` reranker. Ollama with  `mxbai-emded-large:latest` + `cross-encoder/ms-marco-MiniLM-L-6-v2`. I can't run a stronger reranker.

**Comment by tjbck** [comment]
@mbelur milvus issue is not related to this issue #5622

**Comment by tjbck** [comment]
In case someone comes across this in the future, this issue is most likely caused by a version mismatch between an external chromadb and internal open-webui chromadb package version, I was able to reproduce the issue with the latest chromadb image with the current main. After the version bump in the latest dev, I'm no longer able to reproduce the issue.

**Comment by mcsim78** [comment]
I am using version 0.3.30 and encountering an issue when importing a very large Markdown file (27MB, over 100K lines). The problem occurs regardless of the database I use (I've tried both ChromaDB and Milvus). The file's syntax is correct, and splitting it into smaller files is not an option because search performance becomes significantly slower after the import. Interestingly, if I change the file extension to `.txt`, the import process works without any issues.

The problem seems to be related to the `UnstructuredMarkdownLoader` from the `langchain_community.document_loaders` package.

```python
loader = UnstructuredMarkdownLoader(file_path)
data = loader.load()
```

The loader takes a long time to process the file and then returns an empty content:

```python
page_content='' metadata={'source': 'collected_content.md'}
```

**Reply by soymh** [reply]
> I am using version 0.3.30 and encountering an issue when importing a very large Markdown file (27MB, over 100K lines). The problem occurs regardless of the database I use (I've tried both ChromaDB and Milvus). The file's syntax is correct, and splitting it into smaller files is not an option because search performance becomes significantly slower after the import. Interestingly, if I change the file extension to `.txt`, the import process works without any issues.
> 
> The problem seems to be related to the `UnstructuredMarkdownLoader` from the `langchain_community.document_loaders` package.
> 
> ```python
> loader = UnstructuredMarkdownLoader(file_path)
> data = loader.load()
> ```
> 
> The loader takes a long time to process the file and then returns an empty content:
> 
> ```python
> page_content='' metadata={'source': 'collected_content.md'}
> ```

thank you sir
i faced exactly this empty content json!

but wondering what would be the solution?
I've tried both docker dev branch and main branch's pip install,and also tried Changing chromad version to v0.5.5 in docker, and nothing worked;and also you mentioned it's about the function,not chromadb.

it would be so nice if someone finds the solution;

it's totally weird why the main issue is closed whilst we do still have this problem in both dev and main!

**Comment by S1M0N38** [comment]
I was able to reproduce the issue #5736 
I'm running open-webui with Docker with the main tag (0.3.30).
I've set `CHROMA_HTTP_HOST` and `CHROMA_HTTP_PORT` to a running instance of chromadb (0.5.11).

---

I've then downgraded chromadb to 0.5.5 (thesame version used by open-webui 0.3.30), and the error disappeared.

**Reply by soymh** [reply]
Hi sir
I have to mention that the chromadb default version in main pip package install is already v0.5.5 and the problem is still there...

**Reply by lachlanholland1** [reply]
i am having this issue also

**Reply by justinh-rahb** [reply]
0.3.32 is out now and expects the _latest_ Chroma version, not 0.5.5. That workaround was valid for 0.3.30.

**Comment by kim-gtek** [comment]
I am still getting this error 
**`Collection file-b2e4c1f1-569c-4c41-b237-54b2c6965ed8 does not exist.`**

And when I ask a model about the document I get
**`The context appears unreadable or of poor quality.`**

Im using default openwebui setup with openai text-embedding-3-small.
I have reset Vector Storage and Reset Upload directory.

**Reply by bgeneto** [reply]
Despite several `Collection file-UUID does not exist.` error messages like that (when uploading new docs), I can query RAG successfully with v0.3.35. But I had to reset my vector db... This log message is annoying after all...

**Reply by kim-gtek** [reply]
I have checked in the DB, the collection exists, the files are attributed to it, yet the error persists.

When I add a file to a model I get : `The context appears unreadable or of poor quality`

**Reply by Szymok** [reply]
I have the same issue, uploading of the files works correctly but i can not ask any questions about them. Models are not able to fetch the collections.

**Comment by kim-gtek** [comment]
I had my chunking settings wrong, down bottom. Set to 1000 and 100 and it
works now! I dont know why these unset. Personal error or during the update.
